libdevcore/Worker.cpp

```diff
@@ -48,9 +48,13 @@ void Worker::startWorking()
 			while (m_state != WorkerState::Killing)
 			{
 				WorkerState ex = WorkerState::Starting;
-				bool ok = m_state.compare_exchange_strong(ex, WorkerState::Started);
+				{
+					// When somebody waits for this change on m_state, they'll give me the lock saying 'm_state_notifier.wait(l)'
```

Comment:
I'd just say that it's condition_variable-related mutex, maybe it's clearer...

Reply:
OK.